examples/chip-tool/commands/pairing/PairingCommand.h

```diff
@@ -69,19 +93,51 @@ class PairingCommand : public Command, public chip::Controller::DevicePairingDel
     void OnPairingComplete(CHIP_ERROR error) override;
     void OnPairingDeleted(CHIP_ERROR error) override;
 
+    /////////// DeviceAddressUpdateDelegate Interface /////////
+    void OnAddressUpdateComplete(NodeId nodeId, CHIP_ERROR error) override;
+
+    /////////// Network Commissioning Callbacks /////////
+    static void OnDefaultFailureResponse(void * context, uint8_t status);
+    static void OnAddNetworkResponse(void * context, uint8_t errorCode, uint8_t * debugText);
+    static void OnEnableNetworkResponse(void * context, uint8_t errorCode, uint8_t * debugText);
+
 private:
     CHIP_ERROR RunInternal(NodeId remoteId);
     CHIP_ERROR Pair(NodeId remoteId, PeerAddress address);
     CHIP_ERROR PairWithoutSecurity(NodeId remoteId, PeerAddress address);
     CHIP_ERROR Unpair(NodeId remoteId);
 
+    void InitCallbacks();
+    void ReleaseCallbacks();
+    CHIP_ERROR SetupNetwork();
+    CHIP_ERROR AddNetwork(PairingNetworkType networkType);
+    CHIP_ERROR AddThreadNetwork();
+    CHIP_ERROR AddWiFiNetwork();
+    CHIP_ERROR EnableNetwork();
+    CHIP_ERROR UpdateNetworkAddress();
+
     const PairingMode mPairingMode;
+    const PairingNetworkType mNetworkType;
     Command::AddressWithInterface mRemoteAddr;
+    NodeId mRemoteId;
     uint16_t mRemotePort;
+    uint64_t mFabricId;
     uint16_t mDiscriminator;
     uint32_t mSetupPINCode;
+    char * mOperationalDataset;
     char * mSSID;
     char * mPassword;
 
+    chip::Callback::Callback<NetworkCommissioningClusterAddThreadNetworkResponseCallback> * mOnAddThreadNetworkCallback;
+    chip::Callback::Callback<NetworkCommissioningClusterAddWiFiNetworkResponseCallback> * mOnAddWiFiNetworkCallback;
+    chip::Callback::Callback<NetworkCommissioningClusterEnableNetworkResponseCallback> * mOnEnableNetworkCallback;
+    chip::Callback::Callback<DefaultFailureCallback> * mOnFailureCallback;
     ChipDeviceCommissioner mCommissioner;
+    ChipDevice * mDevice;
+    chip::Controller::NetworkCommissioningCluster mCluster;
+
+    // In theory the endpoint should be 0, or alternatively it should be found using the descriptor
+    // cluster.
+    // But at the moment, endpoint starts from 1.
+    chip::EndpointId mEndpointId = 1;
```

Comment:
Looking at #6000 I guess `1` should be replaced by `0` if #6000 lands before this one.